src/M6Web/Component/RedisMock/RedisMock.php

```diff
@@ -332,6 +332,40 @@ public function sismember($key, $member)
 
     // Lists
 
+    public function llen($key)
+    {
+        if (!isset(self::$data[$key]) || $this->deleteOnTtlExpired($key)) {
+            return $this->returnPipedInfo(0);
+        }
+
+        return $this->returnPipedInfo(count(self::$data[$key]));
+    }
+
+    public function lindex($key, $index)
+    {
+        if (!isset(self::$data[$key]) || $this->deleteOnTtlExpired($key)) {
+            // Doc (http://redis.io/commands/lindex) : "When the value at key is not a list, an error is returned."
+            // but what is "an error" ?
+            return $this->returnPipedInfo(null);
+        }
+
+        $size = count(self::$data[$key]);
+
+        // Empty index or out of range
+        if ($size == 0 || abs($index) > $size) {
```

Comment:
I meant `if ($size == 0 || abs($index) >= $size)` (greater **or equal**), in order to discard case where `$index == $size` (and `$index == -$size`). But if you keep the `isset` test of line 362 the behavior will be ok since `self::$data[$key][$size]` will never be defined